docs/content/concepts/assets/software-defined-assets.mdx

```diff
@@ -65,6 +65,24 @@ def downstream_asset(upstream_asset):
 
 The [explicit dependencies](#explicit-dependencies) example covers an alternative way to specify asset dependencies without needing to match argument names to upstream asset names.
 
+### Graph-backed assets
+
+[Basic software-defined assets](#a-basic-software-defined-asset) can only produce one data artifact. If your use case requires an asset that produces more than one data artifact, you can use graph-backed assets.
+
+This approach to asset creation allows you to define a graph of ops which can produce one or multiple assets. **Note**: To use graphs to create an asset, the graph **must return an output**.
+
+To define a graph-backed asset, use the `from_graph` attribute on the `AssetsDefinition` object:
+
+<TODO />
```

Comment:
@clairelin135 would you mind providing example code here? Potential use case:
* First op fetches data from an external data source, maybe a chunk of message from slack
* Second op takes the data and stores it in the warehouse

Here's an example I started building for the migration guide: https://github.com/dagster-io/dagster/pull/8188/files#diff-b85f343582e338e839522ba09bbe85dc3357fc3be9cfa49390b84f2e9d728764R7. It might be a little heavier than we need.

Reply:
@erinkcochran87 Here's an example we can put here (I just included the whole file and marked the example with comments:
```from dagster import op, graph, AssetsDefinition, AssetGroup, ResourceDefinition
from pandas import DataFrame
from unittest import mock


def create_db_connection():
    return "yay"


# start example


@op(required_resource_keys={'slack'})
def fetch_files_from_slack(context) -> DataFrame:
    files = context.resources.slack.files_list(channel="#random")
    return DataFrame(
        [
            {
                "id": file.get("id"),
                "created": file.get("created"),
                "title": file.get("title"),
                "permalink": file.get("permalink"),
            }
            for file in files
        ]
    )


@op
def store_files(files):
    return files.to_sql(name="slack_files", con=create_db_connection())


@graph
def store_slack_files_in_sql():
    store_files(fetch_files_from_slack())


graph_asset = AssetsDefinition.from_graph(store_slack_files_in_sql)

# end example

slack_mock = mock.MagicMock()

store_slack_files = AssetGroup(
    [graph_asset], resource_defs={"slack": ResourceDefinition.hardcoded_resource(slack_mock)}
).build_job("store_slack_files")
```

Reply:
Great - thanks @clairelin135 ! I'll add this in.